### PR TITLE
[automodel] deprecate global_batch_size dataset argument

### PR DESCRIPTION
**Pull request by akoumpa**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Removing deprecated argument

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/gpt/data/hf_dataset.py [hunk 1]
```diff
@@ -203,7 +203,6 @@ class HFDatasetDataModule(pl.LightningDataModule):
         persistent_workers (bool, optional): Whether to keep worker threads alive between epochs. Defaults to True.
         seq_length (int, optional): Maximum sequence length for tokenized inputs. Defaults to 1024.
         micro_batch_size (int, optional): Batch size per device. Defaults to 2.
-        global_batch_size (int, optional): Total batch size across all devices. Defaults to 2.
```
Comment (yuanzhedong) [inline nemo/collections/llm/gpt/data/hf_dataset.py:206]: Wondering why we remove gbs instead of ga? Since gbs is more use-friendly.

File: examples/llm/peft/automodel.py [hunk 2]
```diff
@@ -57,7 +57,6 @@ def formatting_prompts_func(example):
         split="train",
         micro_batch_size=batch_size,
         pad_token_id=tokenizer.eos_id or 0,
-        global_batch_size=batch_size,
         pad_seq_len_divisible=16 if fp8 else None,  # FP8 training requires seq length to be divisible by 16.
```
Comment (athitten) [inline examples/llm/peft/automodel.py:60]: Curious why `datamodule.map(` below harcodes `batch_size=2`

File: nemo/collections/llm/gpt/data/hf_dataset.py (hunk 1 above)
Comment (akoumpa) [reply nemo/collections/llm/gpt/data/hf_dataset.py:206]: Thanks @yuanzhedong , the goal here is to remove an argument that was not being used and was causing confusion. Gradient accumulation is not part of `HFDatasetDataModule` constructor argument list.

File: examples/llm/peft/automodel.py (hunk 2 above)
Comment (akoumpa) [reply examples/llm/peft/automodel.py:60]: the `batch_size` passed to `datamodule.map` controls the preprocessing's batch-size not the training batch size. I haven't experimented too much with the value.

Comment (yuanzhedong) [reply examples/llm/peft/automodel.py:60]: Thanks, my bad, I was confused with the one in automodel.py

File: nemo/collections/llm/gpt/data/hf_dataset.py (hunk 1 above)
Comment (yuanzhedong) [reply nemo/collections/llm/gpt/data/hf_dataset.py:206]: Thanks, my bad, was thinking the one in automodel.py

**Review by yuanzhedong — DISMISSED** [review]

File: nemo/collections/llm/gpt/data/hf_dataset.py [hunk 3]
```diff
@@ -297,7 +295,6 @@ def __init__(
         self.persistent_workers = persistent_workers
         self.seq_length = seq_length
         self.micro_batch_size = micro_batch_size
-        self.global_batch_size = global_batch_size
```
Comment (athitten) [inline nemo/collections/llm/gpt/data/hf_dataset.py:300]: @akoumpa another thing to be mindful of is other places that use self.global_batch_size. For ex:  [nemo/collections/llm/api.py#L1229](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/llm/api.py#L1229)

Comment (akoumpa) [reply nemo/collections/llm/gpt/data/hf_dataset.py:300]: Thanks @athitten
